Snapshot-Pushback is broken

See: https://github.com/kadai-io/kadai/actions/runs/18968785932/job/54172058738

```bash
Run ci/change_version.sh -i -m .
+ git config --global user.email ***
+ git config --global user.name ***
+ git checkout master
Switched to a new branch 'master'
M	ci/kadai-sonar-test-coverage/pom.xml
M	common/kadai-common-data/pom.xml
M	common/kadai-common-logging/pom.xml
M	common/kadai-common-security/pom.xml
M	common/kadai-common-test/pom.xml
M	common/kadai-common/pom.xml
M	common/pom.xml
M	history/kadai-loghistory-provider/pom.xml
M	history/kadai-simplehistory-provider/pom.xml
M	history/kadai-simplehistory-rest-spring/pom.xml
M	history/pom.xml
+ git add './*pom.xml'
M	lib/kadai-core-test/pom.xml
M	lib/kadai-core/pom.xml
M	lib/kadai-spring-example/pom.xml
M	lib/kadai-spring/pom.xml
M	lib/kadai-test-api/pom.xml
M	lib/pom.xml
M	pom.xml
M	rest/kadai-rest-spring-example-boot/pom.xml
M	rest/kadai-rest-spring-example-common/pom.xml
M	rest/kadai-rest-spring-test-lib/pom.xml
M	rest/kadai-rest-spring/pom.xml
M	rest/pom.xml
M	routing/dmn-xlsx-converter/pom.xml
M	routing/kadai-routing-rest/pom.xml
M	routing/kadai-spi-routing-dmn-router/pom.xml
M	routing/pom.xml
M	web/pom.xml
branch 'master' set up to track 'origin/master'.
++ increment_version 11.1.0
++ [[ ! 11.1.0 =~ [0-9]+\.[0-9]+\.[0-9]+ ]]
++ echo 11.1.1
+ git commit -m 'Updated poms to version 11.1.1-SNAPSHOT'
[master de8d3c7f7] Updated poms to version 11.1.1-SNAPSHOT
 28 files changed, 28 insertions(+), 28 deletions(-)
+ git push
remote: error: GH013: Repository rule violations found for refs/heads/master.        
remote: Review all repository rules at https://github.com/kadai-io/kadai/rules?ref=refs%2Fheads%2Fmaster        
remote: 
remote: - Changes must be made through a pull request.        
remote: 
To https://github.com/kadai-io/kadai
 ! [remote rejected]     master -> master (push declined due to repository rule violations)
error: failed to push some refs to 'https://github.com/kadai-io/kadai'
Error: Process completed with exit code 1.
```

--- a/.github/workflows/continuous-integration.yml
+++ b/.github/workflows/continuous-integration.yml
@@ -684,7 +684,7 @@ jobs:
         with:
           fetch-depth: 0 # necessary for push back
           token: ${{ secrets.GITHUB_TOKEN }}
-      - name: Update version to next snapshot and push back
+      - name: Update version to next snapshot and create PR
         run: |
           ci/change_version.sh -i -m .
           ci/commitPoms.sh

--- a/.github/workflows/continuous-integration.yml
+++ b/.github/workflows/continuous-integration.yml
@@ -689,8 +689,6 @@ jobs:
           ci/change_version.sh -i -m .
           ci/commitPoms.sh
         env:
-          GH_EMAIL: ${{ secrets.GH_EMAIL }}
-          GH_USERNAME: ${{ secrets.GH_USERNAME }}
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - name: Cancel workflow
         if: failure()

--- a/.github/workflows/continuous-integration.yml
+++ b/.github/workflows/continuous-integration.yml
@@ -677,6 +677,7 @@ jobs:
     needs: [ release_artifacts ]
     permissions:
       contents: write
+      pull-requests: write
     steps:
       - name: Git checkout
         uses: actions/checkout@v5
@@ -690,6 +691,7 @@ jobs:
         env:
           GH_EMAIL: ${{ secrets.GH_EMAIL }}
           GH_USERNAME: ${{ secrets.GH_USERNAME }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - name: Cancel workflow
         if: failure()
         uses: andymckay/cancel-action@0.5

--- a/ci/commitPoms.sh
+++ b/ci/commitPoms.sh
@@ -12,8 +12,6 @@ set -e # fail fast
 #H
 #H Requirements:
 #H   current commit is a HEAD commit
-#H   GH_USERNAME - github username / displayname (for git config)
-#H   GH_EMAIL - github email address (for git config)
 #H   GH_TOKEN - GitHub token for authentication
 #H   GITHUB_REF (format refs/tags/v[0-9]+\.[0-9]+\.[0-9]+)
 # Arguments:
@@ -57,8 +55,8 @@ function main() {
       exit 1
     fi
     set -x
-    git config --global user.email "$GH_EMAIL"
-    git config --global user.name "$GH_USERNAME"
+    git config --global user.email "github-actions[bot]@users.noreply.github.com"
+    git config --global user.name "github-actions[bot]"
 
     #commit all poms
     git checkout "$branch"

--- a/ci/commitPoms.sh
+++ b/ci/commitPoms.sh
@@ -37,7 +37,7 @@ function increment_version() {
 
 function main() {
   [[ "$1" == '-h' || "$1" == '--help' ]] && helpAndExit 0
-  [[ -z "$GH_EMAIL" || -z "$GH_USERNAME" || -z "$GH_TOKEN" ]] && helpAndExit 1
+  [[ -z "$GH_TOKEN" ]] && helpAndExit 1
   if [[ "$GITHUB_REF" =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
     #check if tagged commit is a head commit of any branch
     commit=$(git ls-remote -q -t origin | grep "$GITHUB_REF" | cut -c1-40)

--- a/ci/commitPoms.sh
+++ b/ci/commitPoms.sh
@@ -14,6 +14,7 @@ set -e # fail fast
 #H   current commit is a HEAD commit
 #H   GH_USERNAME - github username / displayname (for git config)
 #H   GH_EMAIL - github email address (for git config)
+#H   GH_TOKEN - GitHub token for authentication
 #H   GITHUB_REF (format refs/tags/v[0-9]+\.[0-9]+\.[0-9]+)
 # Arguments:
 #   $1: exit code
@@ -38,7 +39,7 @@ function increment_version() {
 
 function main() {
   [[ "$1" == '-h' || "$1" == '--help' ]] && helpAndExit 0
-  [[ -z "$GH_EMAIL" || -z "$GH_USERNAME" ]] && helpAndExit 1
+  [[ -z "$GH_EMAIL" || -z "$GH_USERNAME" || -z "$GH_TOKEN" ]] && helpAndExit 1
   if [[ "$GITHUB_REF" =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
     #check if tagged commit is a head commit of any branch
     commit=$(git ls-remote -q -t origin | grep "$GITHUB_REF" | cut -c1-40)
@@ -61,12 +62,24 @@ function main() {
 
     #commit all poms
     git checkout "$branch"
+
+    NEW_VERSION=$(increment_version "${GITHUB_REF##refs/tags/v}")-SNAPSHOT
+    BRANCH_NAME="chore/version-bump-${NEW_VERSION}-$(date +%Y%m%d-%H%M%S)"
+
+    git checkout -b "$BRANCH_NAME"
     git add "./*pom.xml"
     for file in "$@"; do
       [[ -n "$file" ]] && git add "$file"
     done
-    git commit -m "Updated poms to version $(increment_version "${GITHUB_REF##refs/tags/v}")-SNAPSHOT"
-    git push
+    git commit -m "Updated poms to version ${NEW_VERSION}"
+    git push origin "$BRANCH_NAME"
+
+    echo "Creating pull request for version bump..."
+    gh pr create \
+      --base "$branch" \
+      --title "chore: version bump to ${NEW_VERSION}" \
+      --body "Automated version update after release. All tests have passed in the release workflow." \
+      --label "dependencies,automated"
   else
     echo "Nothing to push - this is not a release!"
   fi

--- a/ci/commitPoms.sh
+++ b/ci/commitPoms.sh
@@ -77,7 +77,7 @@ function main() {
       --base "$branch" \
       --title "chore: version bump to ${NEW_VERSION}" \
       --body "Automated version update after release. All tests have passed in the release workflow." \
-      --label "dependencies,automated"
+      --label "dependencies"
   else
     echo "Nothing to push - this is not a release!"
   fi

